server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeValueHealth.java

```diff
@@ -0,0 +1,316 @@
+    }
+
+    public void setRelocatingShards(int relocatingShards) {
+        this.relocatingShards = relocatingShards;
+    }
+
+    public List<String> getNodeList() {
+        return nodeList;
+    }
+
+    public void setNodeList(List<String> nodeList) {
+        this.nodeList = nodeList;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(name);
+        out.writeVInt(activeShards);
+        out.writeVInt(initializingShards);
+        out.writeVInt(relocatingShards);
+        out.writeVInt(unassignedShards);
+        out.writeVInt(nodes);
+        out.writeDouble(weight);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(getName());
+        builder.field(ACTIVE_SHARDS, getActiveShards());
+        builder.field(INITIALIZING_SHARDS, getInitializingShards());
+        builder.field(RELOCATING_SHARDS, getRelocatingShards());
+        builder.field(UNASSIGNED_SHARDS, getUnassignedShards());
+        builder.field(NODES, getNodes());
+        builder.field(WEIGHTS, getWeight());
+        builder.endObject();
+        return builder;
+    }
+
+    void computeAttributeValueLevelInfo(ClusterState clusterState, boolean displayUnassignedShardLevelInfo, int shardsPerAttributeValue) {
+        // computing nodes info
+        nodes = nodeList.size();
+
+        // computing shards into
+        setShardLevelInfo(clusterState, displayUnassignedShardLevelInfo, shardsPerAttributeValue);
+
+        // compute weight info
+        setWeightInfo(clusterState);
+    }
+
+    private void setShardLevelInfo(ClusterState clusterState, boolean displayUnassignedShardLevelInfo, int shardsPerAttributeValue) {
+
+        for (String nodeId : nodeList) {
+            RoutingNode node = clusterState.getRoutingNodes().node(nodeId);
+            activeShards += node.numberOfShardsWithState(ShardRoutingState.STARTED);
+            relocatingShards += node.numberOfShardsWithState(ShardRoutingState.RELOCATING);
+            initializingShards += node.numberOfShardsWithState(ShardRoutingState.INITIALIZING);
+        }
+
+        // computing unassigned shards info
+        if (displayUnassignedShardLevelInfo) {
+            int unassignedShardsPerAttribute = shardsPerAttributeValue - getActiveShards() - getInitializingShards();
+            setUnassignedShards(unassignedShardsPerAttribute);
+        } else {
+            setUnassignedShards(-1);
+        }
+    }
+
+    private void setWeightInfo(ClusterState clusterState) {
+        WeightedRoutingMetadata weightedRoutingMetadata = clusterState.getMetadata().weightedRoutingMetadata();
+        double attributeWeight = 1.0;
+        if (weightedRoutingMetadata != null) {
+            WeightedRouting weightedRouting = weightedRoutingMetadata.getWeightedRouting();
+            attributeWeight = weightedRouting.weights().getOrDefault(name, 1.0);
```

Comment:
Not sure if we should default to 1 if no weights specified

Reply:
Then this would mean that all values are active and no specific parameter is set